cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_train_dense_default_impl.i

```diff
@@ -948,15 +957,55 @@ public:
 //////////////////////////////////////////////////////////////////////////////////////////
 // ClassificationTrainBatchKernel
 //////////////////////////////////////////////////////////////////////////////////////////
-template <typename algorithmFPType, CpuType cpu>
-services::Status ClassificationTrainBatchKernel<algorithmFPType, defaultDense, cpu>::compute(
+template <typename algorithmFPType, Method method, CpuType cpu>
+services::Status ClassificationTrainBatchKernel<algorithmFPType, method, cpu>::compute(
     HostAppIface * pHostApp, const NumericTable * x, const NumericTable * y, const NumericTable * w, decision_forest::classification::Model & m,
     Result & res, const decision_forest::classification::training::Parameter & par)
 {
     ResultData rd(par, res.get(variableImportance).get(), res.get(outOfBagError).get(), res.get(outOfBagErrorPerObservation).get());
-    services::Status s = computeImpl<algorithmFPType, cpu, daal::algorithms::decision_forest::classification::internal::ModelImpl,
-                                     TrainBatchTask<algorithmFPType, defaultDense, cpu> >(
-        pHostApp, x, y, w, *static_cast<daal::algorithms::decision_forest::classification::internal::ModelImpl *>(&m), rd, par, par.nClasses);
+    services::Status s;
+    dtrees::internal::FeatureTypes featTypes;
+    DAAL_CHECK(featTypes.init(*x), ErrorMemoryAllocationFailed);
+    dtrees::internal::IndexedFeatures indexedFeatures;
+    if (method == hist)
```

Comment:
[Minor] This code will be compiled for hist & defultDense method, this can be avoided by using specialization + This is more consistent with the rest of the algorithms:

```c++
template <typename algorithmFPType, CpuType cpu>
services::Status ClassificationTrainBatchKernel<algorithmFPType, hist, cpu>::compute(
...
template <typename algorithmFPType, CpuType cpu>
services::Status ClassificationTrainBatchKernel<algorithmFPType, defaultDense, cpu>::compute(
...
```